### ISSUE TITLE
Add an option to pass environment variables to the default job

### ISSUE THREAD

**Issue by filipnavara** [dotnet/performance#1045]
When testing against the mono/netcore runtime we often want to run the actual benchmark with specific runtime configuration. This is achieved through setting the `MONO_ENV_OPTIONS` environment variable. However, we also want to use different `MONO_ENV_OPTIONS` when running `dotnet restore` or `dotnet build` for performance reasons. It would be nice to allow some way to inject the environment variables but only for the actul benchmark job.

What I currently use is the following hack:
https://github.com/filipnavara/performance/commit/d1768c7fdf52cc5905bdefce25964c08ba90f46d

/cc @EgorBo

**Comment by adamsitnik** [comment]
Do I understand correctly that you would like to have the possibility to specify two different set of env vars, one for `dotnet build|restore` executed by BDN and another for the benchmark process?

**Comment by filipnavara** [comment]
> Do I understand correctly that you would like to have the possibility to specify two different set of env vars, one for `dotnet build|restore` executed by BDN and another for the benchmark process?

That's correct. However, the ones for `dotnet build|restore` can be set in the context of the outer process so there's no need for a mechanism to set the inside BDN.

**Comment by adamsitnik** [comment]
Ok, so the thing that you are currently doing looks good to me (you are using the right API for the job).

Does it mean that you would like to somehow pass the env vars via console line arguments?

**Comment by EgorBo** [comment]
> Does it mean that you would like to somehow pass the env vars via console line arguments?

Would be nice! 🙂
